sidecar/app/query/base.py

```diff
@@ -174,3 +166,44 @@ def memory_rss_usage(self) -> int:
 
 
 QueryTypeT = TypeVar("QueryTypeT", bound=Query)
+
+
+class MaxQueriesRunningError(Exception):
+    pass
+
+
+@dataclass
+class QueryManager:
+    """
+    The QueryManager allows for a fixed number of queries to run at once,
+    and stores those queries in a dictionary.
+
+    Accessing running queries allows the finish and kill methods to be called
+    from another caller (typically a route handler in the HTTP layer.
+    """
+
+    max_parallel_queries: int = field(init=True, repr=False, default=1)
+    running_queries: dict[str, Query] = field(
+        init=False, repr=True, default_factory=dict
+    )
+
+    def get_from_query_id(self, cls, query_id: str) -> Optional[Query]:
```

Comment:
Why have `self` and `cls` in the same function?

Reply:
`cls` is poorly named, let me fix that. it's meant to be the class of the Query that we are getting.

Reply:
Which are the different classes of Query?